### ISSUE TITLE
Update existing managed nodes with Latest Linux AMI

### ISSUE THREAD

**Issue by tahirali-endurance** [weaveworks/eksctl#2291]
In eksctl, how can we upgrade an existing managed worker nodes to use Latest Linux AMI.  Based on this documentation https://eksctl.io/usage/cluster-upgrade/, do we have to create a new managed node group even for updating existing node groups? In our scenario, we are not autoscaling worker node groups but rather trying to patch them with the latest AMI.

*What help do you need?**
How to patch worker nodes for Latest Linux AMI

**Comment by michaelbeaumont** [comment]
Hi, does the documentation at [upgrading managed nodegroups](https://eksctl.io/usage/eks-managed-nodes/#upgrading-managed-nodegroups) apply to your situation?

**Comment by tahirali-endurance** [comment]
Based on this documentation, I ran the command

```bash
eksctl upgrade nodegroup --name=<node group name> --cluster=<cluster name>
```

This command took some time and finished the update without any errors. After that, when I checked AWS EKS console, and checked the **Compute** tab, it still shows the **_Update Now_** link. Does that mean update is not complete or does AWS console update nodes differently than eksctl?

**Comment by tahirali-endurance** [comment]
I checked the current node group name using eksctl and I found different from what is showing up in AWS EKS console

```bash
eksctl get nodegroup --cluster=<cluster name>

ng-v4 
```
It is showing ng-v4. While, AWS EKS Console is showing ng-v3.

**Comment by michaelbeaumont** [comment]
@tahirali-endurance I'm not sure why you have `ng-v3` and `ng-v4` because I don't know anything about your config or how to reproduce your situation.
It may be that "Update now" is visible for a number of reasons, what versions of kubernetes and which ami is running?

**Comment by martina-if** [comment]
Hi @tahirali-endurance, is it possible that you have two clusters with the same name in different regions and eksctl is looking in one region and the console in a different one?

eksctl should show the same name for the nodegroups.

**Comment by tahirali-endurance** [comment]
It was my mistake, I was checking updates in different AWS account. The node group name remains same after AMI updates. But when we updated the nodes with ```eksctl upgrade nodegroup ```, the AWS console still showed up the updates are available.

**Comment by martina-if** [comment]
@tahirali-endurance what version of eksctl are you using?

**Comment by tahirali-endurance** [comment]
I am not sure exactly as we always download the latest version eksctl in the pipeline. I guess it was wither 0.20 or 0.21.0.

**Comment by martina-if** [comment]
Hi @tahirali-endurance, the only thing I can think of that can cause this is the known lag between CloudFormation and the publishing of new AMIs but that was fixed in version 0.20.0. We don't know what the criteria in AWS console is to show that notification so I'm not sure how we can help. Is this a big issue for your team? Can you try to reproduce it with eksctl 0.20 or above?

**Comment by tahirali-endurance** [comment]
For now, you can close the issue. In our case, we recreated the cluster.
